Cython/Includes/cpython/dict.pxd

```diff
@@ -1,13 +1,40 @@
 from .object cimport PyObject
 
 cdef extern from "Python.h":
+    # On Python 2, PyDict_GetItemWithError is called _PyDict_GetItemWithError
+    """
+    #if PY_MAJOR_VERSION <= 2
+    #define PyDict_GetItemWithError _PyDict_GetItemWithError
+    #endif
+    """
+    # Macro defined in pyport.h
+    ctypedef unsigned long uint64_t
 
     ############################################################################
     # 7.4.1 Dictionary Objects
     ############################################################################
 
-    # PyDictObject
-    #
+    ctypedef struct PyDictEntry:
+        Py_ssize_t me_hash
+        PyObject* me_key
+        PyObject* me_value
+
+    ctypedef struct PyDictKeysObject
+
+    ctypedef class __builtin__.dict [object PyDictObject]:
```

Comment:
I think this might have an impact on the return type of the functions defined here that are declared as returning `dict`. We need to make sure that Cython still considers it the builtin `dict` type and nothing else, so that we don't loose optimisations or add stupid type checks by accident.

Reply:
Find a way to test how it behaves :)